src/components/widgets/DateWidget.js

```diff
@@ -22,9 +12,9 @@ function DateWidget({
     <input type="date"
       id={id}
       className="form-control"
-      value={fromJSONDate(value)}
+      value={typeof value === "undefined" ? "" : value}
       required={required}
-      onChange={(event) => onChange(toJSONDate(event.target.value))} />
+      onChange={(event) => onChange(event.target.value)} />
```

Comment:
Could you explain why we no longer need `toJSONDate` and `fromJSONDate`? It seems like we have the same types of events and they should have the same kinds of values. Are we changing our contract or am I missing something fundamental?